core/src/main/java/org/apache/gravitino/storage/relational/mapper/provider/base/ModelMetaBaseSQLProvider.java

```diff
@@ -134,4 +134,11 @@ public String deleteModelMetasByLegacyTimeline(
         + ModelMetaMapper.TABLE_NAME
         + " WHERE deleted_at > 0 AND deleted_at < #{legacyTimeline} LIMIT #{limit}";
   }
+
+  public String updateModelLatestVersion(@Param("modelId") Long modelId) {
+    return "UPDATE "
+        + ModelMetaMapper.TABLE_NAME
+        + " SET model_latest_version = model_latest_version + 1"
+        + " WHERE model_id = #{modelId} AND deleted_at = 0";
```

Comment:
I think it's better to add the `model_latest_version = xxx` condition in the WHERE clause, we should only update the expected version number. If there are concurrent updates, this may cause the version number to increase incorrectly.

Reply:
I see, let me change the code.

Reply:
I think typically it will not have the concurrent updates problem for a single server Gravitino, looks like it only happens when we have multiple Gravitino instances. I think using `update...set...where` may not really solve the problem, a possible solution is to use transactions to handle this problem.

Reply:
Yeah, I think a single instance based on local lock can solve this problem, but our internal production environment is multi-instance load balancing, and updating without specifying the version may be a potential problem for our scenario. In particular, this SQL will provide an independent method, which may be a potential risk if used in some custom development. WDYT？

Reply:
I change to use a transaction to handle this, can you please review again?